test/functional/feature_evm_contracts.py

```diff
@@ -60,7 +62,50 @@ def setup(self):
         )
         self.nodes[0].generate(1)
 
-    def should_create_contract(self):
+    def generate_contract(self, node: TestNode, num_functions: int, contract_name: str):
+        contract_start = """
+pragma solidity ^0.8.0;\n
+contract {} {{ \n
+    """.format(
+            contract_name
+        )
+
+        function_template = lambda index: """
+    function func{}() public pure returns(uint256) {{
+        return {};
+    }}""".format(
+            index, index
+        )
+
+        contract_end = """
+}"""
+
+        list_sig = []
+        contract_body = ""
+
+        for i in range(0, num_functions):
+            func_sig = "func${}()".format(i)
+            sig_hash = node.w3.keccak(text=func_sig)[:4]
+            if sig_hash in list_sig:
+                continue
+            list_sig.append(sig_hash)
+            contract_body += function_template(i)
+
+        utf8SourceCode = contract_start + contract_body + contract_end
+
+        compiled_output = compile_source(
+            source=utf8SourceCode,
+            output_values=["abi", "bin"],
+            solc_version="0.8.20",
+        )
```

Comment:
I think it would be better to extend [`EVMContract`](https://github.com/DeFiCh/ain/blob/0b2b7f7eeff05a41bf942e21b73826bef19b0b6a/test/functional/test_framework/evm_contract.py#L7) to have a new `from_str` function instead of calling `solcx` directly.